index.js

```diff
@@ -29,9 +32,22 @@ class Main {
       .catch((err) => { throw new Error(`${INIT_EXPORTER_ERR_MSG}${err.message}`); });
   }
 
+  initLastProcessedPosition(lastRecoveredPosition) {
```

Comment:
This logic used to be part of the worker, what's the gain in moving it here? It seems like half the logic has remained there and half is here.